lib/ui/painting.dart

```diff
@@ -1418,6 +1418,44 @@ class Canvas extends NativeFieldWrapperClass2 {
   /// entire group can be made transparent using the [saveLayer]'s paint.
   ///
   /// Call [restore] to pop the save stack and apply the paint to the group.
+  ///
+  /// ## Using saveLayer with clips
+  ///
+  /// When a rectanglular clip operation (from [clipRect]) is not axis-aligned
+  /// with the display, or when the clip operation is not rectalinear (e.g.
+  /// because it is a rounded rectangle clip created by [clipRRect] or an
+  /// arbitrarily complicated path clip created by [clipPath]), the edge of the
+  /// clip needs to be anti-aliased.
+  ///
+  /// If two draw calls overlap at the edge of such a clipped region, without
+  /// using [saveLayer], the first drawing will be anti-aliased with the
+  /// background first, and then the second will be anti-aliased with the result
+  /// of blending the first drawing and the background. On the other hand, if
+  /// [saveLayer] is used immediately after establishing the clip, the second
```

Comment:
Oh my.  Sounds like I've misunderstood how this API works.  You might want to check call sites like https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/rendering/object.dart#L358 and https://github.com/flutter/engine/blob/master/flow/layers/clip_rrect_layer.cc#L50 which sound like they're wrong.